src/NewUser.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from "react";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import FetchedUserData from './FetchedUserData'
+import UserCard from './UserCard'
+import Button from '@material-ui/core/Button';
+
+const END_POINT = "https://randomuser.me/api/?results=5";
+
+function NewUser() {
+  const [loading, setLoading] = useState(false);
+  const [loaded,setLoaded] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [userList, setUserList] = useState([]);
+
+  const fetchUser = async () => {
+    setLoading(true);
+    setLoaded(false);
+    const response = await fetch(END_POINT);
+    response
+      .json()
+      .then((res) => {
+        setUserList(res.results);
+        setLoading(false);
+        setLoaded(true);
+      })
+      .catch((error) => {
+        setHasError(true);
```

Comment:
missing `setLoading(false)` here, so when you will hit error, it will still be in loading state.
for not repeating you can also do `.finally(() => setLoading(false))`